File: day10/crt.pl
#!/usr/bin/env perl
use v5.36;
use Path::Tiny;

my @cmds = path($ARGV[0])->lines({chomp => 1});

my $X = 1;
my $cycle = 1;
my $ip = 0;
my $skip = 1;
my $strengths = 0;
my @crt = (' ') x 240;

while (1) {
    $strengths += $cycle * $X if $cycle % 40 == 20;
    $crt[$cycle-1] = '#' if $X-1 <= ($cycle-1) % 40 <= $X+1;
    my @cmd = split ' ', $cmds[$ip];
    if ($cmd[0] eq 'addx') {
        if ($skip) {
            $ip--;
            $skip = 0;
        } else {
            $X += $cmd[1];
            $skip = 1;
        }
    }
    $cycle++;
    $ip++;
    last if $ip >= @cmds;
}

say "Part 1 $strengths";
say "Part 2";
for (my $i = 0; $i < 240; $i += 40) {
    say @crt[$i..$i+39];
}

File: day10/crt.py
from sys import argv

# read in the program with a one-liner
with open(argv[1]) as f:
    cmds = [line.rstrip().split(' ') for line in f]

X = 1
cycle = 1
ip = 0
skip = True
strengths = 0
while True:
    if cycle % 40 == 20:
        strengths += cycle * X
    cmd = cmds[ip]
    if (cmd[0] == 'addx'):
        if skip:
            skip = False
            ip -= 1
        else:
            X += int(cmd[1])
            skip = True
    cycle += 1
    ip += 1
    if ip >= len(cmds):
        break

print('Part 1', strengths)
